x-pack/plugins/fleet/server/services/actions/actions.ts

```diff
@@ -0,0 +1,283 @@
+  esClient: ElasticsearchClient,
+  actionIds: string[]
+): Promise<{ items: FleetActionRequest[]; total: number }> => {
+  try {
+    const getActionsResponse = await esClient.search({
+      index: AGENT_ACTIONS_INDEX,
+      from: 0,
+      size: ES_SEARCH_LIMIT,
+      query: {
+        bool: {
+          filter: [
+            {
+              terms: {
+                action_id: actionIds,
+              },
+            },
+          ],
+        },
+      },
+    });
+
+    const actions = getActionsResponse.hits.hits.reduce<FleetActionRequest[]>((acc, hit) => {
+      if (hit._source) {
+        acc.push(hit._source as FleetActionRequest);
+      }
+      return acc;
+    }, []);
+
+    return {
+      items: actions,
+      total: actions.length,
+    };
+  } catch (getActionsByIdError) {
+    throw new FleetActionsError(
+      `Error getting action: ${getActionsByIdError.message}`,
+      getActionsByIdError
+    );
+  }
+};
+
+export const getActionsWithKuery = async (
+  esClient: ElasticsearchClient,
+  kuery: string
+): Promise<{ items: FleetActionRequest[]; total: number }> => {
+  try {
+    const query = toElasticsearchQuery(fromKueryExpression(kuery));
+    const getActionSearchResponse = await esClient.search({
+      index: AGENT_ACTIONS_INDEX,
+      from: 0,
+      size: ES_SEARCH_LIMIT,
+      query,
+    });
+
+    const actions = getActionSearchResponse.hits.hits.reduce<FleetActionRequest[]>((acc, hit) => {
+      if (hit._source) {
+        acc.push(hit._source as FleetActionRequest);
+      }
+      return acc;
+    }, []);
+
+    return {
+      items: actions,
+      total: actions.length,
+    };
+  } catch (getActionSearchError) {
+    throw new FleetActionsError(
+      `Error getting actions with kuery: ${getActionSearchError.message}`,
+      getActionSearchError
+    );
+  }
+};
+
+export const getActionResultsByIds = async (
+  esClient: ElasticsearchClient,
+  actionIds: string[]
+): Promise<{ items: FleetActionResult[]; total: number }> => {
+  try {
+    const getActionsResultsResponse = await esClient.search({
+      index: AGENT_ACTIONS_RESULTS_INDEX,
+      from: 0,
+      size: ES_SEARCH_LIMIT,
+      query: {
+        bool: {
+          filter: [
+            {
+              terms: {
+                action_id: actionIds,
+              },
+            },
+          ],
+        },
+      },
+    });
+    const actionsResults = getActionsResultsResponse.hits.hits.reduce<FleetActionResult[]>(
+      (acc, hit) => {
+        if (hit._source) {
+          acc.push(hit._source as FleetActionResult);
+        }
+        return acc;
+      },
+      []
+    );
+
+    return {
+      items: actionsResults,
+      total: actionsResults.length,
+    };
+  } catch (getActionByIdError) {
+    throw new FleetActionsError(
+      `Error getting action results: ${getActionByIdError.message}`,
+      getActionByIdError
+    );
+  }
+};
+
+export const getActionResultsWithKuery = async (
```

Comment:
Similarly to the actions, do we know what are the use cases of using kuery to filter action results?

Reply:
@juliaElastic So I dug around in osquery/endpoint plugins and I don't see any `kuery` searches for either `.fleet-actions` or `.fleet-action-results`. Should we just exclude these methods altogether?

Reply:
If not used, we can remove. What about usages in this comment? https://github.com/elastic/kibana/pull/158833#discussion_r1227900803